doc/wg/core/notes/core-notes-2023-03-31.md

```diff
@@ -0,0 +1,110 @@
+ * Amit: It has been. What are those worries?
+ * Alyssa: Mutex required a constructor, so I needed to use OnceCell for it.
+ * Amit: Mutex new() is const as of Rust 1.63
+ * Alyssa: There does exist a thread-local struct library. But it requires stuff I don't want to use.
+ * Amit: So back to my earlier suggestion, we could in Tock have a thread-local construct, which looks like the macro from the standard library maybe, which is essentially a no-op but is not a no-op in host emulation. Or maybe it does let us have some restrictions, and enforces no access to shared states in interrupt handlers. So this could address the software engineering need of having to declare variables in different ways on host platforms.
+ * Alyssa: Not sure that would work. I see a lot of possible engineering issues. Very tricky to do correctly.
+ * Alyssa: How do you write a thread-local without a thread ID?
+ * Amit: One way of doing it, a platform-specific implementation, you know that the implementation is just a global. There's nothing. But the implementation on host-emulation platforms with threads would use the standard library one maybe.
+ * Alyssa: How is the swap out done?
+ * Amit: I don't know. Not sure if "whether std is available" is something we can check?
+ * Alyssa: That would be neat
+ * Amit: I've seen crates do this explicitly by having the std feature, but that's a little more poisonous. Probably have to pass it around everywhere.
+ * Alyssa: I like the idea of exposing whether you want something thread-local or not. And giving that a shared interface. I still don't see how we'd be able to construct a thread-local without a macro. Seems necessary for how Rust does it.
+ * Alyssa: So if SyncCell new used a macro for a constructor, is that okay?
+ * Amit: I'm not sure I'm following.
+ * Alyssa: If you made sync cells with a macro, is that fine?
+ * Phil: I'd need to see the code. No immediate red flag, but I'd have to see some examples to know.
+ * Alyssa: Do you have any better ideas for determining whether the host platform can support threading? Is there a global Tock flag we could create or that already exists that guarantees single-threading?
+ * Phil: Having always lived in a single-threaded world, I don't know.
+ * Alyssa: So we'd need some way to detect and add something to Tock to determine it.
+ * Alyssa: Because we have a fixed set of platforms, we check the platform now. But that's not scalable to Tock in the same way.
+ * Phil: In terms of C, there are compiler flags for this. Whether you're freestanding or not. I don't know of Rust equivalents. I'd guess that it doesn't because of how fundamental multithreading is.
+ * Alyssa: There's something in the target for the toolchain. But I don't think it's available to compiled programs.
+ * Johnathan: We're actually the only Tock project I know of that's single-threaded. Even the embedded stuff assumes threading in most cases.
```

Comment:
`Tock` should be `Rust` here.

Reply:
Hehehe. The statement is still true both ways!

Reply:
```suggestion
 * Johnathan: We're actually the only Rust project I know of that's single-threaded. Even the embedded stuff assumes threading in most cases.
```